client/src/main/scala/com.crobox.clickhouse/stream/ClickhouseSink.scala

```diff
@@ -49,4 +61,69 @@ object ClickhouseSink extends LazyLogging {
       .mergeSubstreams
       .toMat(Sink.ignore)(Keep.right)
   }
+
+  def toSink(config: Config, client: ClickhouseClient, indexerName: Option[String] = None)(
+      implicit ec: ExecutionContext,
+      settings: QuerySettings = QuerySettings()
+  ): Sink[TableOperation, Future[Done]] = {
+    val indexerGeneralConfig = config.getConfig("crobox.clickhouse.indexer")
+    val mergedIndexerConfig = indexerName
+      .flatMap(
+        theIndexName =>
+          if (indexerGeneralConfig.hasPath(theIndexName))
+            Some(indexerGeneralConfig.getConfig(theIndexName).withFallback(indexerGeneralConfig))
+          else None
+      )
+      .getOrElse(indexerGeneralConfig)
+    val batchSize     = mergedIndexerConfig.getInt("batch-size")
+    val flushInterval = mergedIndexerConfig.getDuration("flush-interval").getSeconds.seconds
+    Flow[TableOperation]
+      .groupBy(Int.MaxValue, _.table)
+      .groupedWithin(batchSize, flushInterval)
+      .mapAsync(mergedIndexerConfig.getInt("concurrent-requests"))(operations => {
+        val table = operations.head.table
+        logger.debug(
+          s"Executing ${operations.size} operations on table: $table. Group Within: ($batchSize - $flushInterval)"
+        )
+
+        insertOp(operations.collect { case o: Insert                     => o }, client)
+          .flatMap(_ => optimizeOp(operations.collect { case o: Optimize => o }, client))
+          .map(_ => operations)
+      })
+      .mergeSubstreams
+      .toMat(Sink.ignore)(Keep.right)
+  }
+
+  private def insertOp(ops: Seq[Insert], client: ClickhouseClient)(
+      implicit ec: ExecutionContext,
+      settings: QuerySettings = QuerySettings()
+  ): Future[Iterable[String]] =
+    Future.sequence(ops.groupBy(_.table).map { t =>
+      val insertQuery = s"INSERT INTO ${t._1} FORMAT JSONEachRow"
```

Comment:
Try to keep condense! (but still readable of course!)

No need to create a new val insertQuery. Directly put the declartion at line 106.